### DISCUSSION TITLE
Is there a javascript example (not typescript) based on hono framework ?

### DISCUSSION THREAD

**Discussion by gouthy** [honojs/hono#1048] [General]
Hello, 
I am still a javascript noob and trying to understand how the hono web framework works. I am trying the implement toy example in the documentation in javascript. I get the following error "No fetch handler defined and no upstream to proxy to specified. Make sure you're calling addEventListener("fetch", ...)." . 
I know eventually, I have to learn typescript but as suggested by many having a knowledge of javascript would be useful for typescript.

Thanks

**Comment by yusukebe** [comment]
Hi @gouthy !

If you want to use `addEventListener("fetch", ...)` you should utilize the `app.fire()` method.

However, even if you are a JavaScript beginner, I recommend learning TypeScript. TypeScript is a superset of JavaScript, so becoming familiar with TypeScript also means mastering JavaScript. In other words, you can learn both JavaScript and TypeScript simultaneously. Fortunately, there's no need to transpile when using Wrangler, Bun, or Deno.

Don't be afraid to use TypeScript!
